### PR TITLE
context: add X-Frame-Options header 

### LINKED ISSUE #6409: Response `X-Frame-Options` in HTTP header
Example: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/internal/cli/http.go#L150:19

Basically add a line below this one:

https://github.com/gogs/gogs/blob/428276d7a8d73ad360ae10e7fdc01e5754e48c8b/internal/context/context.go#L291

Credit to _Faizan Ahmed <ethicalhacker218@gmail.com>_

### PR DESCRIPTION
**Pull request by matheusmosca**
Closes #6409 

I just added the X-Frame-Options header.

### CODE REVIEW

**Review by unknwon — APPROVED** [review]
Thank you! 

Would you mind also adding an entry to the [CHANGELOG.md](https://github.com/gogs/gogs/blob/main/CHANGELOG.md) under **Fixed**?

**Comment by matheusmosca** [comment]
> Thank you!
> 
> Would you mind also adding an entry to the [CHANGELOG.md](https://github.com/gogs/gogs/blob/main/CHANGELOG.md) under **Fixed**?

No problem :thumbsup:

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ All notable changes to Gogs are documented in this file.
 
 ### Fixed
 
+- Add `X-Frame-Options` header [#6409](https://github.com/gogs/gogs/issues/6409)
```
Comment (unknwon) [inline CHANGELOG.md:20]:
```suggestion
- Add `X-Frame-Options` header to prevent Clickjacking. [#6409](https://github.com/gogs/gogs/issues/6409) 
```

**Comment by unknwon** [comment]
Thanks again! Merging once CI passed.

**Pull request merged**